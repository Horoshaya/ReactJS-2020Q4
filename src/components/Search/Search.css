.form {
  display: flex;
}

.input {
  flex-basis: 80%;
  padding: 15px;
  font-size: 18px;
  color: #ffffff;
  border: none;
  border-radius: 5px;
  background: #232323b5;
  outline: none;
  transition-duration: 0.4s;
}

.input:hover {
  background: #232323e8;
}

.button {
  flex-basis: 20%;
  padding: 15px;
  margin-left: 5px;
  font-size: 18px;
  color: #ffffff;
  background: #f65261;
  border: none;
  border-radius: 5px;
  transition-duration: 0.5s;
  cursor: pointer;
}

.button:hover {
  background: #f1d6d6c7;
  color: #f65261;
}
