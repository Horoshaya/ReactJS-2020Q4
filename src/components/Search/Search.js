import React from 'react';
import { useHistory } from 'react-router-dom';
import styles from './Search.css';

const Search = () => {
  const history = useHistory();

  const handleSubmit = (event) => {
    event.preventDefault();
    history.push({
      pathname: '/search',
      search: `${event.target.value}`,
    });
  };

  return (
    <>
      <input
        onKeyUp={handleSubmit}
        className={styles.input}
        type="text"
        name="title"
        placeholder="What do you want to watch?"
      />
      <button className={styles.button} onClick={handleSubmit}>
        Search
      </button>
    </>
  );
};

export default Search;
