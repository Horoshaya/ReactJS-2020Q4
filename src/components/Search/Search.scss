.form {
  display: flex;
}

.input {
  flex-basis: 70%;
  padding: 15px;
  font-size: 18px;
  border: none;
  border-radius: 5px;
  background: #232323ce;
}

.button {
  flex-basis: 10%;
  padding: 15px;
  margin-left: 5px;
  font-size: 18px;
  color: #ffffff;
  background: #f65261;
  border: none;
  border-radius: 5px;
}
