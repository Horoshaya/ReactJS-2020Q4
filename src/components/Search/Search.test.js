import React from 'react';
import rerender from 'react-test-renderer';
import Search from './Search';
import { shallow } from 'enzyme';

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('when Search', () => {
  test('then snapshot created', () => {
    const component = rerender.create(<Search />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('then snapshot created', () => {
    const component = shallow(<Search />);

    const input = component.find('input');
    input.value = 'test';
    const button = component.find('button');
    button.simulate('click', {
      preventDefault: () => {},
      target: {
        value: 'test',
      },
    });
    expect(mockHistoryPush).toHaveBeenCalledWith({
      pathname: '/search',
      search: 'test',
    });
  });
});
