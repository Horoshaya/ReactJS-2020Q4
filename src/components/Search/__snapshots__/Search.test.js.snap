// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Search then snapshot created 1`] = `
Array [
  <input
    name="title"
    onKeyUp={[Function]}
    placeholder="What do you want to watch?"
    type="text"
  />,
  <button
    onClick={[Function]}
  >
    Search
  </button>,
]
`;
